src/main/java/net/countercraft/movecraft/combat/utils/MathHelper.java

```diff
@@ -0,0 +1,49 @@
+package net.countercraft.movecraft.combat.utils;
+
+public class MathHelper {
+
+    public static double clamp(double value) {
+        // Double.MIN_VALUE represents the lowest POSITIVE double value to match IEEE754 format
+        return clamp(-Double.MAX_VALUE, Double.MAX_VALUE, value);
+    }
+
+    // Same as with doubles!
+    public static float clamp(float value) {
+        return clamp(-Float.MAX_VALUE, Float.MAX_VALUE, value);
+    }
+
+    public static int clamp(int value) {
+        return clamp(Integer.MIN_VALUE, Integer.MAX_VALUE, value);
+    }
+
+    public static double clamp(double min, double max, double value) {
+        if (value > max) {
+            return max;
+        }
+        if (value < min) {
+            return min;
+        }
+        return value;
+    }
+
+    public static int clamp(int min, int max, int value) {
+        if (value > max) {
+            return max;
+        }
+        if (value < min) {
+            return min;
+        }
+        return value;
+    }
+
+    public static float clamp(float min, float max, float value) {
+        if (value > max) {
+            return max;
+        }
+        if (value < min) {
+            return min;
+        }
+        return value;
+    }
```

Comment:
All of this should be replaced with one single utility function in `Directors.java` which clamps an entire vector.

Reply:
It is not director specific code. It is math. Thus one does not expect it to be in DirectorUtils, that'y why it is in a separate MathHelper class cause it makes more sense there logically. I can add a function to clamp an entire vector.

Functions should be where they thematically fit.